File: Python/27.removeElement.py
# 双指针
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i, last = 0, len(nums) - 1
        while i <= last:
            if nums[i] == val:
                nums[i], nums[last] = nums[last], nums[i]
                last -= 1
            else:
                i += 1
        return last + 1

File: Javascript/27.removeElement.js
// 双指针
var removeElement = function(nums, val) {
    let begin = 0, end = nums.length - 1;
    while (begin <= end) {
        if (nums[begin] == val) {
            nums[begin] = nums[end--];
        }
        else {
            begin++;
        }
    }
    return end + 1;
};